Elevel/Application/Features/ApplicationUserFeatures/GetAllApplicationUserQuery.cs

```diff
@@ -0,0 +1,150 @@
+﻿using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using Elevel.Application.Interfaces;
+using Elevel.Application.Pagination;
+using Elevel.Domain;
+using Elevel.Domain.Models;
+using MediatR;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Elevel.Application.Features.ApplicationUserFeatures
+{
+    public class GetAllApplicationUserQuery
+    {
+
+        public class Request : PagedQueryBase, IRequest<Response>
+        {
+            public string LastName { get; set; }
+            public string FirstName { get; set; }
+            public string Email { get; set; }
+        }
+
+        public class Handler : IRequestHandler<Request, Response>
+        {
+
+            private readonly IMapper _mapper;
+
+            private readonly IApplicationDbContext _context;
+
+            private readonly UserManager<ApplicationUser> _userManager;
+
+            public Handler(IMapper mapper, UserManager<ApplicationUser> userManager)
+            {
+                _mapper = mapper;
+                _userManager = userManager;
+            }
+
+            public async Task<Response> Handle(Request request, CancellationToken cancellationToken)
+            {
+
+                IQueryable<ApplicationUser> users;
+
+                if (!string.IsNullOrWhiteSpace(request.LastName) 
+                    && !string.IsNullOrWhiteSpace(request.FirstName) 
+                    && !string.IsNullOrWhiteSpace(request.Email))
+                {
+                    users = _userManager.Users.AsNoTracking().Where(x =>
```

Comment:
you can have in 47 row 
IQueryable<ApplicationUser> users = _userManager.Users.AsNoTracking();

and in 53 and next rows just use users = users.Where to avoid code duplication